File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  const numsRecord: Record<number, number> = {};
  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    const complement = target - num;
    if (numsRecord[complement] !== undefined) {
      return [numsRecord[complement], i];
    }
    numsRecord[num] = i;
  }
  throw new Error('No two sum solution found');
}

File: 1-two-sum/two_sum.rb
def two_sum(nums, target)
  nums_record = {}
  nums.each_with_index do |num, i|
    complement = target - num
    if nums_record.key?(complement)
      return [nums_record[complement], i]
    end
    nums_record[num] = i
  end
  raise 'No two sum solution found'
end